algorithms/src/snark/varuna/data_structures/circuit_proving_key.rs

```diff
@@ -27,7 +27,7 @@ use snarkvm_utilities::{
 use std::{cmp::Ordering, sync::Arc};
 
 /// Proving key for a specific circuit (i.e., R1CS matrices).
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, Eq)]
```

Comment:
I'd recommend adding:
```
impl<E: PairingEngine, SM: SNARKMode> Eq for CircuitProvingKey<E, SM> {}
```
next to `PartialEq` for better readability. Otherwise the reader may see only the `PartialEq` impl and not realize `Eq` is also impl'ed.

Reply:
1bc011e53